DailyReportWeb/app/app.js

```diff
@@ -17,9 +17,30 @@
             .when('/', {
                 redirectTo: '/app/welcome'
             })
+            .when('/app/instances/:instanceId/report/:hash', {
+                templateUrl: 'app/report.html',
+                controller: 'ReportCtrl as reportCtrl',
+                resolve: {
+                    report: function ($http, $location, $route) {
+                        var params = { instanceId: $route.current.params.instanceId, hash: $route.current.params.hash };
+
+                        $http.get('api/report/isDashboardAvailable', { params: params })
```

Comment:
Do we need this? Can't we just have the page with a message instead? Like "No content yet" or something like that.

Reply:
Yes, but before that I need to make sure the whole url is ok (instanceId and instance hash). Because I could just guess the instanceId, and just put a random string instead of hash. This method checks if the url is ok.

Reply:
Why not put directly the hash as a query string param? So it needs to match that in order to get you to the dashboard.

Reply:
I don't exactly understand. I still need to make a call to the server to check on that side if the hash is ok. The alternative would be to get the hash before and store it in the js, but that's just bad. Another alternative is to make the get of the dashboard with hash as well, and to throw an error which will come on js and check if it's the one related to the hash. 

Are you talking about the last option?

Reply:
yes, i think he is thinking about the last option. it's kind of weird to have an endpoint that checks if the dashboard is available. when you request the dashboard, if it's not available, just return there a boolean or something

Reply:
Ok, roger.